moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -185,72 +211,46 @@ export default function MapView({ projectName, phaseKey, phaseName }) {
-                !editState.isEditingComponent && (
-                  <>
-                    <ListItem dense>
-                      <Button
-                        size="small"
-                        color="primary"
-                        fullWidth
-                        startIcon={<AddCircleOutlineIcon />}
-                        onClick={onStartCreatingComponent}
-                      >
-                        New Component
-                      </Button>
-                    </ListItem>
-                    <Divider />
-                  </>
-                )}
-              {createState.draftComponent &&
-                createState.isCreatingComponent && (
-                  <DraftComponentListItem
-                    primaryText={createState.draftComponent.component_name}
-                    secondaryText={createState.draftComponent.component_subtype}
-                    onSave={onSaveDraftComponent}
-                    onCancel={onCancelComponentCreate}
-                    saveButtonDisabled={
-                      !createState.draftComponent?.features.length > 0
-                    }
-                    saveButtonText="Save"
-                  />
-                )}
-              {editState.draftEditComponent && editState.isEditingComponent && (
-                <DraftComponentListItem
-                  primaryText={
-                    editState.draftEditComponent?.moped_components
-                      ?.component_name
-                  }
-                  secondaryText={
-                    editState.draftEditComponent?.moped_components
-                      ?.component_subtype
-                  }
-                  onSave={onSaveEditedComponent}
-                  onCancel={onCancelComponentMapEdit}
-                  saveButtonDisabled={!doesDraftEditComponentHaveFeatures}
-                  saveButtonText="Save Edit"
-                />
-              )}
-              {!editState.isEditingComponent &&
-                !createState.isCreatingComponent &&
-                components.map((component) => {
-                  const isExpanded =
-                    clickedComponent?.project_component_id ===
-                    component.project_component_id;
-                  return (
-                    <ComponentListItem
-                      key={component.project_component_id}
-                      component={component}
-                      isExpanded={isExpanded}
-                      setClickedComponent={setClickedComponent}
-                      setIsDeletingComponent={setIsDeletingComponent}
-                      editDispatch={editDispatch}
-                      onClickZoomToComponent={onClickZoomToComponent}
-                      isEditingComponent={editState.isEditingComponent}
-                      isCreatingComponent={createState.isCreatingComponent}
-                    />
-                  );
-                })}
+              <NewComponentToolbar
```

Comment:
I tried to clean up the side bar a bit by composing it out of smaller components. Hopefully, this makes it more straightforward to see the pieces of it.